sdk/client-core/core/src/main/java/io/clientcore/core/http/client/DefaultHttpClient.java

```diff
@@ -418,18 +419,57 @@ private static void buildAndSend(HttpRequest httpRequest, OutputStreamWriter out
 
         /**
          * Reads the response from the input stream and extracts the information needed to construct an instance of
-         * HttpUrlConnectionResponse
+         * Response
          *
          * @param httpRequest The HTTP Request being sent
          * @param reader the input stream from the socket
-         * @return an instance of HttpUrlConnectionResponse
+         * @return an instance of Response
          * @throws IOException If an I/O error occurs
          */
         private static Response<?> buildResponse(HttpRequest httpRequest, BufferedReader reader) throws IOException {
+            // Parse Http response from socket:
+            // Status Line
+            // Response Headers
+            // Blank Line
+            // Response Body
+
+            int statusCode = readStatusCode(reader);
+
+            HttpHeaders headers = readResponseHeaders(reader);
+            HttpHeader contentLengthHeader = headers.get(CONTENT_LENGTH);
+            String line;
+            // read body if present
+            if (headers.getSize() != 0 && contentLengthHeader != null && contentLengthHeader.getValue() != null) {
```

Comment:
This will break if chunked `Transfer-Encoding` is used as it doesn't return a `Content-Length`:

https://developer.mozilla.org/docs/Web/HTTP/Headers/Transfer-Encoding#chunked

Reply:
Adding chunked encoding as a follow up.